### DISCUSSION TITLE
Quotations in Oracle SQL statement cause Ora-942 error (table not found)

### DISCUSSION THREAD

**Discussion by febus** [linq2db/linq2db#3695] [Q&A] [answered]
Hello,
I generated my datacontext using PostgreSQL  schema. Now I  use Oracle database with exact the same db structure.

I can establish db connection using linq2db. The problem is, I always get Ora-942 Table not  found exception.
The reason for it are quatations in generated SQL  statement:

```SQL
SELECT
	t1."idx"
FROM
	"some_sql_table" t1
WHERE
	t1."idx" = 2
```
I get exactly the same error message if I try to execute this SQL statement in sql editor (dbeaver).
To fix this I need only to remove all quotations from above sql statement.
Is there any way to tell linq2db to skip quotations? Any other solutions? With PostgreSQL I didn't notice such problems, it is also still working with PostgreSQL.

**Comment by febus** [comment, marked as answer]
Well, I think I've found a workaround in linq2db sources:
`OracleTools.DontEscapeLowercaseIdentifiers = true;`

After this call all my lowercase identfiers will be not quotated anymore.

**Reply by jods4** [reply]
For reference: Oracle schema is case-sensitive by default.
The "trick" is that unquoted names are automatically transformed to uppercase by Oracle.

So `some_table` actually looks for a case-sensitive object named `SOME_TABLE`.
`"SOME_TABLE"` would work the same.
`"some_table"` on the other hand looks for lowercase `some_table`, which is a different object.

By default linq2db assumes you might be using lowercase Oracle object names, that's why it quotes lowercase identifiers.
The setting you found skips this and is perfect if you are a sane person who only uses uppercase Oracle object names.
